go.mod

```diff
@@ -1,32 +1,58 @@
 module github.com/reddit/baseplate.go
 
-go 1.16
+go 1.17
```

Comment:
can you run `go mod tidy -go=1.17` on this? I think it will add a second block to facilitate dep graph pruning and lazy loading

(it should only need to be run once, I believe)

Reply:
I already did that (it's the same as running `go mod tidy` with a `go 1.17` line in `go.mod`), but merged the two `require` blocks together afterwards.

But looks like 1.17's default behavior is to group indirects together, so I changed `go.mod` file to that.